### PR TITLE
Fixed orbital angular momentum calculation for RTP:

### PR DESCRIPTION
**Pull request by mattiatj**
1. use non-symmetric neighbour list in qs_moments::build_local_magmom_matrix
2. track imaginary part of the density if magnetic moments are requested
3. use imaginary part of the density to calculate the expectation value

### CODE REVIEW

File: src/qs_moments.F [hunk 1]
```diff
@@ -522,24 +519,14 @@ SUBROUTINE build_local_magmom_matrix(qs_env, magmom, nmoments, ref_point, ref_po
                              sphi_b(1, sgfb), SIZE(sphi_b, 1), &
                              0.0_dp, work(1, 1), SIZE(work, 1))
 
-                  IF (iatom <= jatom) THEN
+                  ! IF (iatom <= jatom) THEN
```
Comment (dev-zero) [inline src/qs_moments.F]: please do not include commented out code, and re-indent the following code properly

File: src/qs_moments.F [hunk 2]
```diff
@@ -1495,11 +1482,14 @@ SUBROUTINE qs_moment_locop(qs_env, magnetic, nmoments, reference, ref_point, uni
...
       TYPE(cp_result_type), POINTER                      :: results
       TYPE(dbcsr_p_type), DIMENSION(:), POINTER          :: magmom, matrix_s, moments, rho_ao
+      TYPE(dbcsr_type), POINTER                          :: rho_magmom_ao
       TYPE(dft_control_type), POINTER                    :: dft_control
       TYPE(distribution_1d_type), POINTER                :: local_particles
       TYPE(particle_type), DIMENSION(:), POINTER         :: particle_set
       TYPE(qs_kind_type), DIMENSION(:), POINTER          :: qs_kind_set
       TYPE(qs_rho_type), POINTER                         :: rho
+      ! TYPE(rt_prop_type), POINTER                        :: rtp
+      ! LOGICAL                                            :: imag_density = .FALSE.
```
Comment (dev-zero) [inline src/qs_moments.F]: dito, but ok'ish if you plan to use those with a follow-up commit

File: src/rt_propagation_types.F [hunk 3]
```diff
@@ -97,6 +97,7 @@ MODULE rt_propagation_types
       REAL(KIND=dp)                               :: mixing_factor
       LOGICAL                                     :: mixing
       LOGICAL                                     :: do_hfx
+      LOGICAL                                     :: magnetic
```
Comment (dev-zero) [inline src/rt_propagation_types.F:100]: unused?

File: src/qs_moments.F (hunk 1 above)
Comment (mattiatj) [reply src/qs_moments.F]: Done.

File: src/qs_moments.F (hunk 2 above)
Comment (mattiatj) [reply src/qs_moments.F]: Deleted, actually don't need them

File: src/rt_propagation_types.F (hunk 3 above)
Comment (dev-zero) [reply src/rt_propagation_types.F:100]: uh, my bad, sorry

**Comment by dev-zero** [comment]
you have to run `make pretty` as well

**Comment by dev-zero** [comment]
... and it seems that your changes broke the test with magnetic moments (`H2O-magnetic`)

**Comment by mattiatj** [comment]
In my opinion the regtest H20-magnetic should give zero orbital angular momentum if there is no magnetic field

**Comment by dev-zero** [comment]
Makes sense, and I guess this is why it says in the output:

```
[...]
  Magnetic Dipole Moment [WRONG]
    X=   -0.00851550 Y=    0.00496533 Z=   -0.65955777     Total=      0.65963143
[...]
```

Could you therefore please correct the test value in `tests/QS/regtest-ot-1/TEST_FILES` and remove the `[WRONG]` string in `src/qs_moments.F` since you fixed it?

Might be good if someone else @juerghutter or @marci73 could take a look as well?

**Pull request merged**